### PR TITLE
767 add gateway

### LINKED ISSUE #764: Can't install custom EnvoyFleets
Currently when trying to run the command to install another fleet, it returns the following error: 

```bash
kusk install --name dashboard-envoyfleet --namespace dashboard-kusk-system
  ✔  Looking for Helm...
  ✔  Adding Kubeshop repository...
  ✔  Fetching the latest charts...
 WARNING  WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /Users/aabedraba/Downloads/civo-kusk-demo-kubeconfig
          WARNING: Kubernetes configuration file is world-readable. This is insecure. Location: /Users/aabedraba/Downloads/civo-kusk-demo-kubeconfig
     ERROR:  Installing Kusk Gateway: process error: exit status 1
             output: WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /Users/aabedraba/Downloads/civo-kusk-demo-kubeconfig
             WARNING: Kubernetes configuration file is world-readable. This is insecure. Location: /Users/aabedraba/Downloads/civo-kusk-demo-kubeconfig
             Release "dashboard-envoyfleet" does not exist. Installing it now.
             Error: rendered manifests contain a resource that already exists. Unable to continue with install: ClusterRole "kusk-gateway-api-editor-role" in namespace "" exists and cannot be imported into the current release: invalid ownership metadata; annotation validation error: key "meta.helm.sh/release-name" must equal "dashboard-envoyfleet": current value is "kusk-gateway"; annotation validation error: key "meta.helm.sh/release-namespace" must equal "dashboard-kusk-system": current value is "kusk-system"
```

Could we fix that? 

Furthermore, I think it'll be great if we could have a specific command for it... e.g. 

```
kusk gateway create
```
? Or something in that fashion....

### LINKED ISSUE #767: CLI: kusk cluster add-gateway
Add ability to add custom EnvoyFleet from the CLI as mentioned in #764 

With new install in place this should be much easier to achieve.

@aabedraba we should make this into a wizard. Pick a name, pick namespace, type ClusterIP/Loadbalancer, port and similar.

### PR DESCRIPTION
**Pull request by jasmingacic**
Fixes #767 
Fixes #764

### CODE REVIEW

**Comment by kylehodgetts** [comment]
Why not group this and api under deploy?
```
kusk deploy api ...
kusk deploy gateway...
```

**Comment by jasmingacic** [comment]
@donnalux This is just a preview how the wizard behaves:
image
image
image
image

Nothing is set in stone so we can move it as @kylehodgetts and @aabedraba said above

**Comment by donnalux** [comment]
@jasmingacic proud parent here, I have no feedback, it looks and reads great.

As for the command itself, these are the options mentioned:

kusk gateway add
kusk deploy gateway
kusk cluster gateway add

I quite like "kusk deploy gateway" since it fits nicely with existing pattern.

**Comment by jasmingacic** [comment]
This is missing option to remove the gateway. I will have it added soon

**Comment by jasmingacic** [comment]
image
image
image

File: cmd/kusk/manifests/api_server.yaml [hunk 1]
```diff
@@ -129,4 +129,4 @@ spec:
   selector:
     app.kubernetes.io/instance: kusk-gateway-api
     app.kubernetes.io/name: kusk-gateway-api
-  type: LoadBalancer
+  type: ClusterIP
```
Comment (jasmingacic) [inline cmd/kusk/manifests/api_server.yaml:132]: For some reason I changed this into LoadBalancer in previous commits. Here I reverted it.

**Pull request merged**